include/swift/AST/Decl.h

```diff
@@ -1862,8 +1862,11 @@ class PatternBindingEntry {
 public:
   PatternBindingEntry(Pattern *P, SourceLoc EqualLoc, Expr *E,
                       DeclContext *InitContext)
-      : ThePattern(P), EqualLoc(EqualLoc), InitAndFlags(E, {}),
-        InitContext(InitContext) {}
+      : ThePattern(P), EqualLoc(EqualLoc), InitAndFlags(nullptr, {}),
+        InitContext(InitContext) {
+    // Setting the initializer does some book-keeping, so we do that separately.
+    setInit(E);
```

Comment:
This doesn't seem to be true anymore. Maybe it can go back to the way it was?

Reply:
Ah, yeah. This and the change in ASTVerifier and CodeSynthesis are all just left overs from previous versions. Reverted.